core/amber/src/main/scala/edu/uci/ics/amber/engine/common/AmberConfig.scala

```diff
@@ -105,4 +108,66 @@ object AmberConfig {
   // JDBC configuration
   val jdbcConfig: Config = getConfSource.getConfig("jdbc")
 
+  // Language server configuration
+  val languageServer: String = getConfSource.getString("language-server.default")
+
+  private val logger: Logger = Logger.getLogger(this.getClass.getName)
+
+  //free the port for the server
+  private def releasePort(port: Int): Unit = {
+    val pythonCode =
+      s"""
+         |import os
+         |
+         |def release_port(port):
+         |    cmd_find = 'netstat -aon | findstr %s' % (port)
+         |    print('cmd_find:', cmd_find)
+         |    res = os.popen(cmd_find).read()
+         |    print('res:', res)
+         |    if str(port) and 'LISTENING' in res:
+         |        i = res.index('LISTENING')
+         |        start = i + len('LISTENING') + 7
+         |        end = res.find('\\n', start)
+         |        pid = res[start:end].strip()
+         |        print('pid:', pid)
+         |        cmd_kill = 'taskkill -f -pid %s' % (pid)
+         |        print('cmd_kill:', cmd_kill)
+         |        os.popen(cmd_kill)
+         |    else:
+         |        print('This port is free to use')
+         |
+         |release_port($port)
+       """.stripMargin
+
+    val command = Seq("python", "-c", pythonCode)
+    val exitCode = command.!
+    if (exitCode == 0) {
+      logger.info(s"Successfully free the port: $port")
+    } else {
+      logger.warning(s"fail to free the port: $port")
+    }
+  }
+
+  def startLanguageServer(): Unit = {
+    languageServer match {
+      case "pyright" =>
+        logger.info("Starting Pyright...")
+        releasePort(3000)
+        //Try to run the Pyright language server
+        try {
+          val result =
+            Process("node ../languageServer/startPyright.mjs").run()
+          logger.info("Pyright language server is running on port 3000")
+        } catch {
+          case e: Exception => logger.warning(s"Failed to start Pyright: ${e.getMessage}")
+        }
+      case "pylsp" =>
+        logger.info("Starting Pylsp...")
+        releasePort(3000)
+        val result = Process("pylsp --ws --port 3000").run()
```

Comment:
make sure we are ignoring stdout from this process.  for stderr, we can append it into this process.